media/js/newsletter/form-utils.es6.js

```diff
@@ -16,135 +21,265 @@ const errorList = {
+        return '';
+    },
 
-        if (typeof response !== 'object') {
-            response = JSON.parse(response);
+    /**
+     * Gets the user's UUID token.
+     * Looks to cookie first, falling back to local variable.
+     * Will return an empty string if token is not a valid format.
+     * @returns {String}
+     */
+    getUserToken: () => {
+        let token;
+
+        const cookiesEnabled =
+            typeof Mozilla.Cookies !== 'undefined' && Mozilla.Cookies.enabled();
+
+        if (cookiesEnabled) {
+            token = Mozilla.Cookies.getItem(BASKET_COOKIE_ID);
+        } else {
+            token = _userToken;
         }
 
-        if (response) {
-            if (
-                response.status === 'ok' &&
-                e.target.status >= 200 &&
-                e.target.status < 300
-            ) {
-                successCallback();
-            } else if (response.status === 'error' && response.desc) {
-                errorCallback(response.desc);
+        if (FormUtils.isValidToken(token)) {
+            return token;
+        } else {
+            return '';
+        }
+    },
+
+    /**
+     * Validates the supplied token matches the expected UUID format.
+     * @param {String} token
+     * @returns {Boolean}
+     */
+    isValidToken: (token) => {
+        if (typeof token !== 'string') {
+            return false;
+        }
+
+        const matches = token.match(UUID_REGEX);
+
+        if (matches && matches[0]) {
+            return true;
+        } else {
+            return false;
+        }
+    },
+
+    /**
+     * Returns true if URL string starts with http(s) or a relative path.
+     * @param {String} url
+     * @returns {Boolean}
+     */
+    isWellFormedURL: (url) => {
+        const absolute = /^https?:\/\//;
+        const relative = /^\//;
+        return absolute.test(url) || relative.test(url);
+    },
+
+    /**
+     * Perform an AJAX POST to Basket
+     * @param {String} email
+     * @param {String} params (URI encoded query string)
+     * @param {String} url (Basket API endpoint)
+     * @param {Function} successCallback
+     * @param {Function} errorCallback
+     */
+    postToBasket: (email, params, url, successCallback, errorCallback) => {
+        const xhr = new XMLHttpRequest();
+
+        // Emails used in automation for page-level integration tests
+        // should avoid hitting basket directly.
+        if (email === 'success@example.com') {
+            successCallback();
+            return;
+        } else if (email === 'failure@example.com') {
+            errorCallback();
+            return;
+        }
+
+        xhr.onload = function (e) {
+            let response = e.target.response || e.target.responseText;
+
+            if (typeof response !== 'object') {
+                response = JSON.parse(response);
+            }
+
+            if (response) {
+                if (
+                    response.status === 'ok' &&
```

Comment:
I think we're ok here since `status` is returned by the server, whereas `statusText` is a browser feature (hence why we saw different behavior there in some browsers).